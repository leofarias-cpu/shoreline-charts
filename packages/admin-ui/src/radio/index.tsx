import { Radio as ReakitRadio } from 'reakit/Radio'
import type { RadioStateReturn } from 'reakit/Radio'
import { createComponent, useElement } from '@vtex/admin-ui-react'

import * as style from './Radio.style'

export const Radio = createComponent<typeof ReakitRadio, RadioOptions>(
  (props) => {
    const { state, ...restProps } = props

    return useElement(ReakitRadio, {
      ...restProps,
      ...state,
      baseStyle: style.radioStyle,
    })
  }
)

interface RadioOptions {
  state: RadioStateReturn
}

export { useRadioState, RadioStateReturn } from 'reakit/Radio'

export type RadioProps = React.ComponentPropsWithoutRef<typeof Radio>
