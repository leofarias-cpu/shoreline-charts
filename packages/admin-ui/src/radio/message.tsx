import type { ReactNode } from 'react'
import React from 'react'

import { Set } from '../components/Set'
import { Text } from '../components/Text'

export function Message(props: MessageProps) {
  const { helperText, criticalMessage, tone } = props

  const isCritical = tone === 'critical' && criticalMessage
  const hasMessage = isCritical || helperText

  return hasMessage ? (
    <Set spacing="$xs" orientation="vertical">
      {helperText ? (
        <Text variant="detail" tone="secondary">
          {helperText}
        </Text>
      ) : null}
      {isCritical ? (
        <Text variant="detail" tone="critical">
          {criticalMessage}
        </Text>
      ) : null}
    </Set>
  ) : (
    <></>
  )
}

export interface MessageProps {
  helperText?: ReactNode
  criticalMessage?: ReactNode
  tone?: 'critical' | 'neutral'
}
