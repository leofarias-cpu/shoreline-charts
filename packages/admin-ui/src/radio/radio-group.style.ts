import { styleVariants, style } from '@vtex/admin-ui-core'

export const radioContainerStyles = style({
  display: 'flex',
  alignItems: 'flex-start',
})

export const radioGroupStyles = style({
  display: 'flex',
  flexDirection: 'column',
  '> :not(:last-child)': {
    marginBottom: '$space-2',
  },
})

export const variants = styleVariants({
  orientation: {
    vertical: {
      flexDirection: 'column',
      '> :not(:last-child)': {
        marginBottom: '$space-2',
      },
    },
    horizontal: {
      flexDirection: 'row',
      '> :not(:last-child)': {
        marginRight: '$space-5',
      },
    },
  },
})
