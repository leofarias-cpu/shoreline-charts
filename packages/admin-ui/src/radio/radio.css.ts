import type { Theme } from '@vtex/admin-ui-core'
import { focusVisible, style, csx } from '@vtex/admin-ui-core'
import { get } from '@vtex/admin-ui-util'

export const checkmark = style({
  content: '""',
  position: 'absolute',
  display: 'block',
  borderRadius: '$pill',
  size: '0.5rem',
  bg: '$form.control',
})

export const disabled = style({
  ':disabled': {
    bg: '$disabled',
    border: '$disabled',
    color: '$disabled',
    cursor: 'not-allowed',
    '+ div': {
      label: {
        cursor: 'not-allowed',
        color: '$disabled',
      },
    },
  },
})

export const checked = style({
  ...focusVisible('main'),
  bg: '$form.controlChecked',
  color: '$form.controlChecked',
  border: '$form.controlChecked',
  position: 'relative',

  '&:after': {
    ...checkmark,
    bg: (theme: Theme) => get(theme, 'fg.form.controlChecked', ''),
  },

  ':hover:not(:disabled)': {
    bg: '$form.controlCheckedHover',
    border: '$form.controlCheckedHover',
  },

  '&[disabled]:after': {
    ...checkmark,
    bg: (theme: Theme) => get(theme, 'fg.disabled', ''),
  },

  ...disabled,
})

export const radioTheme = csx({
  ...focusVisible('neutral'),
  size: '1.25rem',
  border: '$form.control',
  appearance: 'none',
  position: 'relative',
  borderRadius: '$pill',
  cursor: 'pointer',
  display: 'flex',
  margin: '$space-0',
  alignItems: 'center',
  justifyContent: 'center',

  ':hover:not(:disabled)': {
    bg: '$form.controlHover',
    border: '$form.controlHover',
  },
  ...disabled,
  ':after': checkmark,
  ':checked': checked,
})

export const labelTheme = csx({
  cursor: 'pointer',
})
