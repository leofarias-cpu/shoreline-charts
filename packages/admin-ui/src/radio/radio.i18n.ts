export const messages = {
  radioGroup: {
    'en-US': { optional: 'Optional' },
    'es-AR': { optional: 'Optional' },
    'fr-FR': { optional: 'Optional' },
    'pt-BR': { optional: 'Opcional' },
    'ja-JP': { optional: 'Optional' },
    'ko-KR': { optional: 'Optional' },
    'it-IT': { optional: 'Optional' },
    'nl-NL': { optional: 'Optional' },
    'ro-RO': { optional: 'Optional' },
    'bg-BG': { optional: 'Optional' },
  },
}
