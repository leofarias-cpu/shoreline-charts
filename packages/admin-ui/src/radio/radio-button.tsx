import { Radio as ReakitRadio } from 'reakit/Radio'
import type { RadioStateReturn } from 'reakit/Radio'

import { createComponent, useElement } from '@vtex/admin-ui-react'

import * as style from './radio.style'

export const RadioButton = createComponent<
  typeof ReakitRadio,
  RadioButtonOptions
>((props) => {
  const { state, ...restProps } = props

  return useElement(ReakitRadio, {
    ...restProps,
    ...state,
    baseStyle: style.radioButtonStyle,
  })
})

export interface RadioButtonOptions {
  state: RadioStateReturn
}

export { RadioStateReturn } from 'reakit/Radio'

export type RadioButtonProps = React.ComponentPropsWithoutRef<
  typeof RadioButton
>
