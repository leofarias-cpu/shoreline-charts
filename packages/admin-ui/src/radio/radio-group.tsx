import type { ReactNode } from 'react'
import React from 'react'
import { createComponent, useElement } from '@vtex/admin-ui-react'
import type { RadioState } from 'ariakit/radio'
import { RadioGroup as AriakitRadioGroup } from 'ariakit/radio'

import {
  FormControl,
  FormControlLabel,
  FormControlMessage,
} from '../form-control'
import { Stack } from '../stack'

export const RadioGroup = createComponent<'fieldset', RadioGroupOptions>(
  (props) => {
    const {
      label,
      helpText,
      errorText,
      children,
      direction,
      state,
      error,
      optional,
      ...restProps
    } = props

    return useElement('fieldset', {
      ...restProps,
      children: (
        <FormControl error={error} optional={optional}>
          <FormControlLabel as="legend">{label}</FormControlLabel>
          <AriakitRadioGroup state={state}>
            <Stack direction={direction} space="$space-4" csx={{ marginY: '$space-1' }}>
              {children}
            </Stack>
          </AriakitRadioGroup>
          <FormControlMessage helpText={helpText} errorText={errorText} />
        </FormControl>
      ),
    });
  }
)

export type RadioGroupProps = React.ComponentPropsWithoutRef<typeof RadioGroup>

export interface RadioGroupOptions {
  /**
   * useRadioState hook return
   */
  state: RadioState
  /**
   * RadioGroup label
   */
  label: ReactNode
  /**
   * Whether is a error field or not
   * @default false
   */
  error?: boolean
  /**
   * RadioGroup children direction
   * @default row
   */
  direction?: 'row' | 'column'
  /**
   * RadioGroup helper text
   */
  helpText?: ReactNode
  /**
   * RadioGroup error text. It appears when error property is set to true.
   */
  errorText?: ReactNode
  /**
   * Whether the field is optional or not
   */
  optional?: boolean
}

export { useRadioState } from 'ariakit/radio'
export type { RadioState } from 'ariakit/radio'
