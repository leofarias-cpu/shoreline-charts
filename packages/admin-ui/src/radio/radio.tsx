import type { ReactNode, Ref } from 'react'
import React, { forwardRef } from 'react'
import { unstable_useId as useId } from 'reakit/Id'

import { Label } from '../label'
import { Stack } from '../stack'
import { Inline } from '../inline'
import type { RadioButtonProps } from './radio-button'
import { RadioButton } from './radio-button'
import { FormControl, FormControlMessage } from '../form-control'
import { labelTheme } from './radio.css'

export const Radio = forwardRef(function Radio(
  props: RadioProps,
  ref: Ref<HTMLInputElement>
) {
  const { label, helpText, id, ...radioButtonProps } = props

  const { id: baseId } = useId({ id })

  return (
    <FormControl>
      <Inline hSpace="$space-2" vSpace="">
        <RadioButton ref={ref} {...radioButtonProps} id={baseId} />
        <Stack space="$space-05">
          <Label htmlFor={baseId} className={labelTheme}>
            {label}
          </Label>
          <FormControlMessage helpText={helpText} />
        </Stack>
      </Inline>
    </FormControl>
  )
})

export type RadioProps = RadioButtonProps & {
  /**
   * Radio label
   */
  label: ReactNode
  /**
   * Helper text
   */
  helpText?: ReactNode
}
