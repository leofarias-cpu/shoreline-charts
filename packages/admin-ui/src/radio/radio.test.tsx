import React from 'react'

import { render, axe, withState } from '../test-utils'
import { Radio, useRadioState } from './radio'

const StatefulRadio = withState(Radio, () =>
  useRadioState({ baseId: 'radio-test' })
)

describe('Radio', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <>
        <StatefulRadio value="unchecked" label="label" />
        <StatefulRadio value="checked" label="label" checked />
        <StatefulRadio value="unchecked disabled" label="label" disabled />
        <StatefulRadio
          value="checked disabled"
          label="label"
          disabled
          checked
        />
      </>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('should not have a11y violations', async () => {
    const { container } = render(<StatefulRadio value="test" label="test" />)

    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
