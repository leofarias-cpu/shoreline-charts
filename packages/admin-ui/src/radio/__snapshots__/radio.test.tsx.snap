// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio should match snapshot 1`] = `
<DocumentFragment>
  .emotion-1 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: -0.0009em;
}

.emotion-1 >span {
  margin-left: 0.5rem;
}

.emotion-0 {
  width: 1.25rem;
  height: 1.25rem;
  padding: 0.3rem;
  background: transparent;
  border: 1px solid #B9B9B9;
  color: #000000;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  position: relative;
  border-radius: 100%;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #DDDDDD;
}

.emotion-0:disabled {
  background: #E6E6E6;
  border: 1px solid #747474;
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:hover {
  background: #E6E6E6;
  border: 1px solid #545454;
}

.emotion-0:after {
  content: "";
  position: absolute;
  display: block;
  border-radius: 100%;
  top: 0.3rem;
  width: 0.5rem;
  height: 0.5rem;
  background: transparent;
}

.emotion-0:checked {
  background: #2953B2;
  color: #FFFFFF;
  border: 1px solid #2953B2;
  position: relative;
}

.emotion-0:checked:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-0:checked:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #BBCBF0;
}

.emotion-0:checked:after {
  content: "";
  position: absolute;
  display: block;
  border-radius: 100%;
  top: 0.3rem;
  width: 0.5rem;
  height: 0.5rem;
  background: #FFFFFF;
}

.emotion-0:checked[disabled]:after {
  content: "";
  position: absolute;
  display: block;
  border-radius: 100%;
  top: 0.3rem;
  width: 0.5rem;
  height: 0.5rem;
  background: #747474;
}

.emotion-0:checked:disabled {
  cursor: not-allowed;
  background: #E6E6E6;
  border: 1px solid #747474;
}

.emotion-0:checked:hover {
  background: #20408C;
  border: 1px solid #20408C;
}

<label
    class="emotion-1"
  >
    <input
      aria-checked="false"
      class="emotion-0"
      id="radio-test-1"
      name="radio-test"
      tabindex="0"
      type="radio"
      value="unchecked"
    />
    <span>
      label
    </span>
  </label>
  .emotion-1 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: -0.0009em;
}

.emotion-1 >span {
  margin-left: 0.5rem;
}

.emotion-0 {
  width: 1.25rem;
  height: 1.25rem;
  padding: 0.3rem;
  background: transparent;
  border: 1px solid #B9B9B9;
  color: #000000;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  position: relative;
  border-radius: 100%;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #DDDDDD;
}

.emotion-0:disabled {
  background: #E6E6E6;
  border: 1px solid #747474;
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:hover {
  background: #E6E6E6;
  border: 1px solid #545454;
}

.emotion-0:after {
  content: "";
  position: absolute;
  display: block;
  border-radius: 100%;
  top: 0.3rem;
  width: 0.5rem;
  height: 0.5rem;
  background: transparent;
}

.emotion-0:checked {
  background: #2953B2;
  color: #FFFFFF;
  border: 1px solid #2953B2;
  position: relative;
}

.emotion-0:checked:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-0:checked:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #BBCBF0;
}

.emotion-0:checked:after {
  content: "";
  position: absolute;
  display: block;
  border-radius: 100%;
  top: 0.3rem;
  width: 0.5rem;
  height: 0.5rem;
  background: #FFFFFF;
}

.emotion-0:checked[disabled]:after {
  content: "";
  position: absolute;
  display: block;
  border-radius: 100%;
  top: 0.3rem;
  width: 0.5rem;
  height: 0.5rem;
  background: #747474;
}

.emotion-0:checked:disabled {
  cursor: not-allowed;
  background: #E6E6E6;
  border: 1px solid #747474;
}

.emotion-0:checked:hover {
  background: #20408C;
  border: 1px solid #20408C;
}

<label
    class="emotion-1"
  >
    <input
      aria-checked="true"
      checked=""
      class="emotion-0"
      id="radio-test-1"
      name="radio-test"
      tabindex="0"
      type="radio"
      value="checked"
    />
    <span>
      label
    </span>
  </label>
  .emotion-1 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: -0.0009em;
}

.emotion-1 >span {
  margin-left: 0.5rem;
}

.emotion-0 {
  width: 1.25rem;
  height: 1.25rem;
  padding: 0.3rem;
  background: transparent;
  border: 1px solid #B9B9B9;
  color: #000000;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  position: relative;
  border-radius: 100%;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #DDDDDD;
}

.emotion-0:disabled {
  background: #E6E6E6;
  border: 1px solid #747474;
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:hover {
  background: #E6E6E6;
  border: 1px solid #545454;
}

.emotion-0:after {
  content: "";
  position: absolute;
  display: block;
  border-radius: 100%;
  top: 0.3rem;
  width: 0.5rem;
  height: 0.5rem;
  background: transparent;
}

.emotion-0:checked {
  background: #2953B2;
  color: #FFFFFF;
  border: 1px solid #2953B2;
  position: relative;
}

.emotion-0:checked:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-0:checked:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #BBCBF0;
}

.emotion-0:checked:after {
  content: "";
  position: absolute;
  display: block;
  border-radius: 100%;
  top: 0.3rem;
  width: 0.5rem;
  height: 0.5rem;
  background: #FFFFFF;
}

.emotion-0:checked[disabled]:after {
  content: "";
  position: absolute;
  display: block;
  border-radius: 100%;
  top: 0.3rem;
  width: 0.5rem;
  height: 0.5rem;
  background: #747474;
}

.emotion-0:checked:disabled {
  cursor: not-allowed;
  background: #E6E6E6;
  border: 1px solid #747474;
}

.emotion-0:checked:hover {
  background: #20408C;
  border: 1px solid #20408C;
}

<label
    class="emotion-1"
  >
    <input
      aria-checked="false"
      aria-disabled="true"
      class="emotion-0"
      disabled=""
      id="radio-test-1"
      name="radio-test"
      style="pointer-events: none;"
      type="radio"
      value="unchecked disabled"
    />
    <span>
      label
    </span>
  </label>
  .emotion-1 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 475;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: -0.0009em;
}

.emotion-1 >span {
  margin-left: 0.5rem;
}

.emotion-0 {
  width: 1.25rem;
  height: 1.25rem;
  padding: 0.3rem;
  background: transparent;
  border: 1px solid #B9B9B9;
  color: #000000;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  position: relative;
  border-radius: 100%;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #DDDDDD;
}

.emotion-0:disabled {
  background: #E6E6E6;
  border: 1px solid #747474;
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:hover {
  background: #E6E6E6;
  border: 1px solid #545454;
}

.emotion-0:after {
  content: "";
  position: absolute;
  display: block;
  border-radius: 100%;
  top: 0.3rem;
  width: 0.5rem;
  height: 0.5rem;
  background: transparent;
}

.emotion-0:checked {
  background: #2953B2;
  color: #FFFFFF;
  border: 1px solid #2953B2;
  position: relative;
}

.emotion-0:checked:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-0:checked:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #BBCBF0;
}

.emotion-0:checked:after {
  content: "";
  position: absolute;
  display: block;
  border-radius: 100%;
  top: 0.3rem;
  width: 0.5rem;
  height: 0.5rem;
  background: #FFFFFF;
}

.emotion-0:checked[disabled]:after {
  content: "";
  position: absolute;
  display: block;
  border-radius: 100%;
  top: 0.3rem;
  width: 0.5rem;
  height: 0.5rem;
  background: #747474;
}

.emotion-0:checked:disabled {
  cursor: not-allowed;
  background: #E6E6E6;
  border: 1px solid #747474;
}

.emotion-0:checked:hover {
  background: #20408C;
  border: 1px solid #20408C;
}

<label
    class="emotion-1"
  >
    <input
      aria-checked="true"
      aria-disabled="true"
      checked=""
      class="emotion-0"
      disabled=""
      id="radio-test-1"
      name="radio-test"
      style="pointer-events: none;"
      type="radio"
      value="checked disabled"
    />
    <span>
      label
    </span>
  </label>
</DocumentFragment>
`;
